woocommerce/woocommerce-ios
# [HACK week] Allow notices to be swiped away

## Description


In the past, I've tried to swipe away our in-app notices, and been annoyed that I couldn't, especially when they're sat in front of other UI which I want to use.

It turns out, they can be dismissed by tapping on them. I didn't know this until I looked at the code.

In the spirit of Helpful Acts of Customer Kindness, I figured that there must be other people like me who never thought to just tap on them, instead of swiping. They slide up from the bottom, it makes sense that they would slide back down again!

### Testing instructions


1. Open the app and get a notice to show. The easiest way I found was to switch stores
2. Swipe down on the notice
3. Observe that it fades away

## Screenshots


https://user-images.githubusercontent.com/2472348/193305000-9b135e9d-979a-4c85-aedf-6f57de07bed1.mp4

## Y u no slide it off!?
Sad, I know. It definitely should slide down, and I got pretty close, even with it as a nice interactive/interruptible animation.

https://user-images.githubusercontent.com/2472348/193308439-63446ff7-afad-432a-8ad0-3e4172258f05.mp4

The changes in this PR took 5 minutes, and I spent about a day on-and-off, trying to get a slide down animation to work. It always had issues: commonly a 20px jump down at the beginning when animating the constraint change (as we do to slide it on.) 

https://user-images.githubusercontent.com/2472348/193307138-e4352643-ea89-420f-8d61-cb49078894d6.mp4

Or it would just immediately disappear.

https://user-images.githubusercontent.com/2472348/193308179-97032abb-54ea-4780-b612-af0a5b5029bb.mp4

Or, if I animated the frame instead of the constraint, it might go up a bit first, then slide down again.

https://user-images.githubusercontent.com/2472348/193307782-b4034a86-71b2-4aa1-8b80-2c820e88da10.mp4

Or even: it would go different directions on iOS 14 compared to iOS 15 and 16 (I don't even know what that was about, it's not like the co-ordinate system was reversed.)

At the end, when I looked in the view hierarchy inspector, I noticed a runtime constraint issue with the labels inside the Notice: height and vertical position being ambiguous. My suspicion is that's causing most of the issues, but I couldn't fix it in the time I had. Maybe next time!

I did get a good refresher on UIView animations though, so that's nice 😄 

---
